Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,326 @@
+    {
+        string? contentType = null;
+
+        if (headers != null && headers.TryGetValue("Content-Type", out var headerContentType))
+        {
+            contentType = headerContentType;
+        }
+        else if (multiValueHeaders != null && multiValueHeaders.TryGetValue("Content-Type", out var multiValueContentType))
+        {
+            contentType = multiValueContentType[0];
+        }
+
+        if (contentType != null)
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.HttpV1)
+            {
+                response.ContentType = "text/plain; charset=utf-8";
+            } 
+            else if (emulatorMode == ApiGatewayEmulatorMode.Rest)
+            {
+                response.ContentType = "application/json";
+            }
+        }
+
+        if (statusCode != 0)
+        {
+            response.StatusCode = statusCode;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.Rest) // rest api text for this message/error code is slightly different
+            {
+                response.StatusCode = 502;
+                response.ContentType = "application/json";
+                var errorBytes = Encoding.UTF8.GetBytes("{\"message\": \"Internal server error\"}");
+                response.Body = new MemoryStream(errorBytes);
+                response.ContentLength = errorBytes.Length;
+            } else
+            {
+                SetInternalServerError(response);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Sets the content type and status code for API Gateway v2 responses.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the content type and status code on.</param>
+    /// <param name="headers">The headers.</param>
+    /// <param name="body">The response body.</param>
+    /// <param name="statusCode">The status code to set.</param>
+    private static void SetContentTypeAndStatusCodeV2(HttpResponse response, IDictionary<string, string>? headers, string? body, int statusCode)
+    {
+        if (headers != null && headers.TryGetValue("Content-Type", out var contentType))
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            response.ContentType = "text/plain; charset=utf-8"; // api gateway v2 defaults to this content type if none is provided
+        }
+
+        if (statusCode != 0)
+        {
+            response.StatusCode = statusCode;
+        }
+        // v2 tries to automatically make some assumptions if the body is valid json
+        else if (IsValidJson(body))
+        {
+            // API Gateway 2.0 format version assumptions
+            response.StatusCode = 200;
+            response.ContentType = "application/json";
```

Comment:
Should you check to see if content type was already set from the headers collection coming back from the Lambda function?

Reply:
ill add an integration test case to confirm what happens in that scenario and update accordingly

Reply:
ive updated my test case  locally. so in the case of inferring the json response, it doesnt use whats in the headers.

```
  [Fact]
  public async Task V2_TestCanInferJsonType2()
  {

      var testResponse = new APIGatewayHttpApiV2ProxyResponse
      {
          Body = "{\"key\" : \"value\"}",
          Headers = new Dictionary<string, string>
              {
                  { "Content-Type", "application/xml" },
              },
      };

      var httpTestResponse = testResponse.ToHttpResponse();
      var actualResponse = await _httpClient.PostAsync(_httpApiV2Url, new StringContent("{\"key\" : \"value\"}"));

      await _helper.AssertResponsesEqual(actualResponse, httpTestResponse);
      Assert.Equal(200, (int)actualResponse.StatusCode);
      Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
      var content = await actualResponse.Content.ReadAsStringAsync();
      Assert.Equal("{\"key\" : \"value\"}", content);
  }
```

this test case passed when i ran it

Reply:
ive added `V2_TestJsonInfersAlwaysUsesApplicationJson`. i believe my current logic is right. the reason being is for this scenario, since status code is 0, api gateway just uses the whole payload as the body response and doesnt consider any lambda provided headers.

Reply:
i updated this logic to just be straightforward - with the results from my manual testing. that if status code is not set, then api gateway http 2.0 will default to application/json

Reply:
i guess what you meant by this (based on your other comment with the binary type for rest api), is that if the user configured custom response headers for this error code that it might be in the response already? ill have to check that

Reply:
so i was thinking about this more with the way the v2 infers things. i think it actually depends on what language the function is written in. Because for .net statusCode is an int, so default value is 0 and its not possible to be null, which is why when you were testing before you got internal server error. but when i was testing in javascript, i excluded the  statuscode from the response object completely, which api gateway was then able to interpret.

i wonder if we should be updating APIGatewayHttpApiV2ProxyResponse.StatusCode to be nullable/check that logic? or should we just actually remove the logic i have here for when the status code is 0, and i can default to internal server error then.

Reply:
i removed the json inference check from here. i think we should add the lambda -> api gateay translation inference in another pr